examples/workforce/workforce.py

```diff
@@ -0,0 +1,57 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from camel.agents.chat_agent import ChatAgent
+from camel.messages.base import BaseMessage
+from camel.tasks.task import Task
+from camel.utils.channel import Channel
+from camel.workforce.unit_workforce import UnitWorkforce
+from camel.workforce.workforce import Workforce
+
+
+def main():
+    human_task = Task(content='develop a python program of investing stock.')
+
+    public_channel = Channel()
+
+    sys_msg_1 = BaseMessage.make_assistant_message(
+        role_name="programmer",
+        content="You are a python programmer.",
+    )
+    sys_msg_2 = BaseMessage.make_assistant_message(
+        role_name="researcher",
+        content="You are good at marketing",
+    )
+    sys_msg_3 = BaseMessage.make_assistant_message(
+        role_name="product owner",
+        content="You are familiar with internet.",
+    )
+    agent_1 = ChatAgent(sys_msg_1)
+    agent_2 = ChatAgent(sys_msg_2)
+    agent_3 = ChatAgent(sys_msg_3)
+
+    unit_workforce_1 = UnitWorkforce(1, 'agent1', agent_1, public_channel)
+    unit_workforce_2 = UnitWorkforce(2, 'agent2', agent_2, public_channel)
+    unit_workforce_3 = UnitWorkforce(3, 'agent3', agent_3, public_channel)
```

Comment:
All the units share the same channel(queue), does it means all the messages in the queue are processed sequentially, even there may be tasks could be done parallel?

Reply:
Tasks in the queue will be distributed to different units. They can process the tasks parallelly.